build/release/teamcity-support.sh

```diff
@@ -105,3 +105,8 @@ verify_docker_image(){
   return $error
 }
 
+
+function is_release_or_master_build(){
+  # On no match, `grep -Eo` returns 1. `|| echo""` makes the script not error.
+  echo "$1" | grep -Eo "^((staging|release|rc)-(v)?[0-9][0-9]\.[0-9](\.0)?).*|master$" || echo ""
```

Comment:
does this need the `-rc` in it?

Reply:
Oh, I see the `rc` at the beginning here. Does this mean the `-rc` isn't being put at the end anymore?

Reply:
`.*` covers all suffixes.

Reply:
Actually, let me simplify it.

Reply:
What do you think about `"^((staging|release|rc)-(v)?[0-9][0-9]\.[0-9]).*|master$"`
Everything that starts with `staging`/`release`/`rc` (we plan to use `rc` in the future), followed by version (with optional `v` prefix - I don't recall exactly, but it was used in the past :/), no matter the suffix, or `master` will match the condition.

Reply:
> Does this mean the `-rc` isn't being put at the end anymore?

Nope. We still have it there for now until all our tooling supports the `rc` prefix.

Reply:
To note, this won't match release tags, for example `v24.1.0`. If that's not needed, then this works.

Reply:
I'm guessing it's not given this is in the make and publish build files. So LGTM assuming this is the case.

Reply:
I don't think we used release tags in the past.